src/components/sbb-expansion-panel/sbb-expansion-panel.tsx

```diff
@@ -0,0 +1,203 @@
+  @Prop() public color: 'white' | 'milk' = 'white';
+
+  /** Whether the panel is expanded. */
+  @Prop({ mutable: true, reflect: true }) public expanded = false;
+
+  /** Whether the panel is disabled, so its expanded state can't be changed. */
+  @Prop({ reflect: true }) public disabled = false;
+
+  /** Whether the panel has no border. */
+  @Prop({ reflect: true }) public borderless = false;
+
+  /** Whether the animations should be disabled. */
+  @Prop({ reflect: true }) public disableAnimation = false;
+
+  /** Emits whenever the sbb-expansion-panel starts the opening transition. */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'will-open',
+  })
+  public willOpen: EventEmitter<void>;
+
+  /** Emits whenever the sbb-expansion-panel is opened. */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'did-open',
+  })
+  public didOpen: EventEmitter<void>;
+
+  /** Emits whenever the sbb-expansion-panel begins the closing transition. */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'will-close',
+  })
+  public willClose: EventEmitter<void>;
+
+  /** Emits whenever the sbb-expansion-panel is closed. */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'did-close',
+  })
+  public didClose: EventEmitter<void>;
+
+  @Element() private _element: HTMLElement;
+
+  @Listen('toggle-expanded')
+  public toggleExpanded(): void {
+    this.expanded = !this.expanded;
+  }
+
+  @Watch('expanded')
+  public onExpandedChange(): void {
+    this._element
+      .querySelector('sbb-expansion-panel-header')
+      .setAttribute('expanded', String(this.expanded));
+    this._element
+      .querySelector('sbb-expansion-panel-content')
+      .setAttribute('expanded', String(this.expanded));
+    this._element.style.setProperty(
+      '--sbb-expansion-panel-content-height',
+      `${this._contentElement.scrollHeight}px`,
+    );
+
+    if (this.expanded) {
+      this.willOpen.emit();
+      this.disableAnimation && this.didOpen.emit();
+    } else {
+      this.willClose.emit();
+      this.disableAnimation && this.didClose.emit();
+    }
+  }
+
+  private _contentElement: HTMLElement;
+  private _transitionEventController: AbortController;
+  private _progressiveId = `-${++nextId}`;
+
+  private _onHeaderSlotChange(event): void {
+    const header = (event.target as HTMLSlotElement)
+      .assignedElements()
+      .find(
+        (e): e is HTMLSbbExpansionPanelHeaderElement => e.tagName === 'SBB-EXPANSION-PANEL-HEADER',
+      );
+    if (!header) {
+      return;
+    }
+
+    header.setAttribute('expanded', String(this.expanded));
+    if (this.disabled) {
+      header.setAttribute('disabled', String(this.disabled));
+    }
+    if (!header.getAttribute('id')) {
+      header.setAttribute('id', `sbb-expansion-panel-header${this._progressiveId}`);
+    }
+
+    const content = this._element.querySelector('sbb-expansion-panel-content');
```

Comment:
I think I would prefer to extract to linking logic to its own method, that checks if both header and content are slotted and is called from both slot change methods.
What do you think?

Reply:
Done in 155f68db